src/services/pallets/PalletsStakingProgressService.ts

```diff
@@ -20,17 +20,27 @@ export class PalletsStakingProgressService extends AbstractService {
 		const [
 			validatorCount,
 			forceEra,
-			eraElectionStatus,
 			validators,
 			{ number },
 		] = await Promise.all([
 			api.query.staking.validatorCount.at(hash),
 			api.query.staking.forceEra.at(hash),
-			api.query.staking.eraElectionStatus.at(hash),
 			api.query.session.validators.at(hash),
 			api.rpc.chain.getHeader(hash),
 		]);
 
+		let eraElectionStatus;
+		try {
+			eraElectionStatus = await api.query.staking.eraElectionStatus.at(hash);
+		} catch {
+			/**
+			 * Runtime v30 and above do not support eraElectionStatus, so use
+			 * a `try` to retrieve the eraElectionStatus, and if were running on a
+			 * runtime less than v30 it will return a successful result. If it doesn't
+			 * we do nothing and let the undefined `eraElectionStatus` dictate the below logic.
+			 */
+		}
```

Comment:
What does `api.query.staking.eraElectionStatus` return on `v30` and beyond? I'd have thought it returns an error (or is even `undefined`?). I think it's better to avoid try-catch when we can, as they risk catching and silencing unexpected errors. So if something like this works, it's better imo:

```suggestion
eraElectionStatus = await api.query.staking.eraElectionStatus.at(hash);
if (!eraElectionStatus) {
  // report warning?
}
```

Reply:
Yea of-course. I think that's totally fair.

To build off your example it would look more like 

```typescript
let eraElectionStatus;
if(api.query.staking.eraElectionStatus) {
    eraElectionStatus = await api.query.staking.eraElectionStatus.at(hash);
}

// Else do nothing, allow `eraElectionStatus` to be undefined, and use a ternary to report it's been Deprecated.
```

But yes I agree this could be much simpler with potentially less overhead via ditching `catch`. The only thing to make sure is we don't check `api.query.staking.eraElectionStatus.at()` because the undefined `at` with the undefined `eraElectionStatus` will cause 500 error.